tests/conformance_dsl/fragment.rs

```diff
@@ -140,6 +141,161 @@ impl TryFrom<Sequence> for Fragment {
     }
 }
 
+// newtype to handle writing an Element, after we check to make sure it's not a symbol that has our
+// special absent symbol sauce.
+#[derive(Debug)]
+pub(crate) struct ProxyElement<'a>(pub &'a Element);
+
+impl<'a> ProxyElement<'a> {
+
+    fn write_struct<V: ValueWriter>(&self, val: &Struct, writer: V) -> ion_rs::IonResult<()> {
+        let annotations: Vec<&Symbol> = self.0.annotations().iter().collect();
+        let annot_writer = writer.with_annotations(annotations)?;
+        let mut strukt = annot_writer.struct_writer()?;
+
+        for (name, value) in val.fields() {
+            match parse_absent_symbol(name.text().unwrap_or("")) {
+                (None, None) => { strukt.write(name, ProxyElement(value))?; }
+                (_, Some(id)) => { strukt.write(RawSymbolRef::from(id), ProxyElement(value))?; },
+                _ => unreachable!(),
+            }
+        }
+        strukt.close()
+    }
+
+    fn write_symbol<V: ValueWriter>(&self, writer: V) -> ion_rs::IonResult<()> {
+        if !self.0.is_null() {
+            let annotations: Vec<&Symbol> = self.0.annotations().iter().collect();
+            let annot_writer = writer.with_annotations(annotations)?;
+            let symbol = self.0.as_symbol().unwrap();
+            match parse_absent_symbol(symbol.text().unwrap_or("")) {
+                (None, None) => annot_writer.write(symbol),
+                (_, Some(id)) => annot_writer.write(RawSymbolRef::from(id)),
+                _ => unreachable!(),
+            }
+        } else {
+            writer.write(self.0)
+        }
+    }
+
+}
+
+impl<T: ion_rs::Decoder> PartialEq<ion_rs::LazyValue<'_, T>> for ProxyElement<'_> {
+    fn eq(&self, other: &ion_rs::LazyValue<'_, T>) -> bool {
```

Comment:
This method has a lot of `unwrap()`s and `unreachable!`s--would you mind adding either comments/`expect()` text/`unreachable!` messages explaining why they shouldn't happen?

A comments pass would also be helpful.

Reply:
I reworked this impl a bit, factored out the symbol comparison to a new `compare_symbols_eq` function that handles the absent/missing symbol notation on both sides, a little more cleanly. I want to pull the equality tests out of the PartialEq implementations, there's too many options and results to handle.